packages/webpack/src/runtime/runtimeBuilder.js

```diff
@@ -16,191 +60,335 @@ module.exports = {
    *   options
    */
   runtimeBuilder({ options }) {
-    const runtimeOptions = {
-      scuttleGlobalThis: options.scuttleGlobalThis,
-      lockdown: options.lockdown,
+    /**
+     * Generates the preamble that caches selected globals to protect runtime
+     * from scuttling. The set of cached globals is limited to ones known to be
+     * used by runtime modules for the sake of bundle size.
+     *
+     * @returns {string}
+     */
+    function getDefensiveCodingPreamble() {
+      const globals = [
+        'location', // AutoPublicPathRuntimeModule.js
+        'setTimeout', // LoadScriptRuntimeModule.js
+        'clearTimeout', // LoadScriptRuntimeModule.js
+        'document', // LoadScriptRuntimeModule.js, AutoPublicPathRuntimeModule.js
+        'trustedTypes', // GetTrustedTypesPolicyRuntimeModule.js
+        'self',
+      ]
+      return `var ${globals.map((g) => `${g} = globalThis.${g}`).join(',')};
+const LOCKDOWN_SHIMS = [];`
     }
 
-    return {
+    /**
+     * Prepares the static shims to be included in the runtime chunk.
+     *
+     * @param {string[]} dependencies - The module specifiers to prepare.
+     * @returns {string} The prepared runtime dependencies source.
+     */
+    function getStaticShims(dependencies) {
       /**
-       * Generates the preamble that caches selected globals to protect runtime
-       * from scuttling. The set of cached globals is limited to ones known to
-       * be used by runtime modules for the sake of bundle size.
+       * Wraps static shim source code to capture the lockdown shim it sets (if
+       * any) Makes LOCKDOWN_SHIMS unreachable in its scope and freezes so no
+       * other runtime modules can add any new shims to it.
        *
-       * @returns {string}
+       * @param {string} source - The source code to wrap.
+       * @returns {string} The wrapped source code.
        */
-      getDefensiveCodingPreamble() {
-        const globals = [
-          'location', // AutoPublicPathRuntimeModule.js
-          'setTimeout', // LoadScriptRuntimeModule.js
-          'clearTimeout', // LoadScriptRuntimeModule.js
-          'document', // LoadScriptRuntimeModule.js, AutoPublicPathRuntimeModule.js
-          'trustedTypes', // GetTrustedTypesPolicyRuntimeModule.js
-          'self',
-        ]
-        return `var ${globals.map((g) => `${g} = globalThis.${g}`).join(',')};`
+      const shimWrap = (source) => `;{
+          let LOCKDOWN_SHIM;((LOCKDOWN_SHIMS)=>{
+            ${source}
+          ;
+          })();
+          if(typeof LOCKDOWN_SHIM === 'function') {
+            LOCKDOWN_SHIMS.push(LOCKDOWN_SHIM)
+          };
+        }`
+
+      const shims = dependencies.map((dep) => {
+        const source = prepareSource(dep)
+        try {
+          new Function(source)
```

Comment:
why is this check needed?

Reply:
It makes sure the JS is correct before concatenation. We don't want it to contain a surplus if closing braces.